### DISCUSSION TITLE
[Fixed] ESP32 compile error: no matching function for call to 'WiFiClass::begin(String&, String&)'

### DISCUSSION THREAD

**Discussion by EmbeddedDevver** [mathieucarbou/MycilaESPConnect#22] [General]
Compilation for ESP32 gives the error 

`no matching function for call to 'WiFiClass::begin(String&, String&)'`

Looking up I see this
`  WiFi.begin(_config.wifiSSID, _config.wifiPassword);`

if I change to 
` WiFi.begin(_config.wifiSSID.c_str(), _config.wifiPassword.c_str());`

Everything runs fine. The WiFi.begin does not accept String only c_str();

Do I use the wrong WiFi begin lib? Or is this an issue with the code :-)

My settings:

```
`[env:esp32-gateway]
platform = espressif32@6.7.0
board = esp32-gateway
framework = arduino
monitor_speed = 115200
monitor_filters = time

build_flags = 
  -Wall -Wextra
  -D CONFIG_ARDUHAL_LOG_COLORS
  -D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_DEBUG
  -D ESPCONNECT_DEBUG
  -D NVS_LOG
  ; -D HTTPCLIENT_NOSECURE
  ; -D ESPCONNECT_NO_MDNS
  ; -D DEBUG_ESP_CORE

lib_deps =
    mathieucarbou/MycilaESPConnect`
```

**Comment by mathieucarbou** [comment]
Hello, you might want to look a bit more and you'll discover that Arduino also provides API for `String`:

```c++
    wl_status_t begin(const String& ssid, const String& passphrase = (const char*)NULL, int32_t channel = 0, const uint8_t* bssid = NULL, bool connect = true) {
        return begin(ssid.c_str(), passphrase.c_str(), channel, bssid, connect);
    }
```

But I added you config to the build: 

https://github.com/mathieucarbou/MycilaESPConnect/commit/560266a66c492e5905f7fb4441fcb6bc2ee38c6b

CI job passed:

https://github.com/mathieucarbou/MycilaESPConnect/actions/runs/9766416658/job/26959326144
